src/evals/data.py

```diff
@@ -0,0 +1,145 @@
+"""
+
+import logging
+
+from composer.utils import MissingConditionalImportError, dist
+
+_glue_task_column_names = {
+    "cola": ("sentence", None),
+    "mnli": ("premise", "hypothesis"),
+    "mrpc": ("sentence1", "sentence2"),
+    "qnli": ("question", "sentence"),
+    "qqp": ("question1", "question2"),
+    "rte": ("sentence1", "sentence2"),
+    "sst2": ("sentence", None),
+    "stsb": ("sentence1", "sentence2"),
+}
+
+_superglue_task_column_names = {
+    "boolq": ("question", "passage"),
+    "cb": ("premise", "hypothesis"),
+    "copa": ("premise", "choice1", "choice2", "question"),
+    "multirc": ("paragraph", "question", "answer"),
+    # "record": ("question1", "question2"), ['passage', 'query', 'entities', 'entity_spans', 'answers', 'idx'
+    "rte": ("premise", "hypothesis"),
+    "wic": (
+        "sentence1",
+        "sentence2",
+    ),  #'word','sentence1'  'sentence2',  'start1',  'start2',  'end1',  'end2',
+    # "wsc": ("sentence1", "sentence2"), #'text','span1_index',  'span2_index',  'span1_text',  'span2_text',
+    # "wsc.fixed": ("sentence1", "sentence2"), #'text','span1_index',  'span2_index',  'span1_text',  'span2_text',
+    # "axb": ("sentence1", "sentence2"),
+    # "axg": ("premise", "hypothesis"),
+}
+
+log = logging.getLogger(__name__)
+
+
+def create_eval_dataset(
+    task: str,
+    tokenizer_name: str,
+    split: str,
+    dataset_name: str,
+    max_seq_length: int = 256,
+    max_retries: int = 10,
+    num_workers: int = 0,
+    dataset_subset: str = None,
+    task_column_names: dict = _glue_task_column_names,
+    tokenize_fn_factory: callable = None,
+):
+    try:
+        import datasets
+        import transformers
+    except ImportError as e:
+        raise MissingConditionalImportError(
+            extra_deps_group="nlp", conda_package="transformers"
+        ) from e
+
+    if task not in task_column_names:
+        raise ValueError(f"task ({task}) must be one of {task_column_names.keys()}")
+
+    if (max_seq_length % 8) != 0:
+        log.warning(
+            "For performance, a max_seq_length as a multiple of 8 is recommended."
+        )
+
+    tokenizer = transformers.AutoTokenizer.from_pretrained(tokenizer_name)  # type: ignore (thirdparty)
+
+    log.info(f"Loading {task.upper()} on rank {dist.get_global_rank()}")
+    download_config = datasets.DownloadConfig(max_retries=max_retries)
+    dataset = datasets.load_dataset(
+        dataset_name,
+        dataset_subset if dataset_subset is not None else task,
+        split=split,
+        download_config=download_config,
+    )
+
+    log.info(f"Starting tokenization by preprocessing over {num_workers} threads!")
+    text_column_names = task_column_names[task]
+
+    if tokenize_fn_factory is None:
+        tokenize_fn_factory = lambda tokenizer, max_seq_length: lambda inp: tokenizer(
+            text=inp[text_column_names[0]],
+            text_pair=(
```

Comment:
a comment explaning that this adds a [SEP] in between is welcome IMO 😄

Reply:
We should also mention that without NSP this token is not pretrained I guess but since we train first on MNLI and then use that checkpoint, we still get meaningful results.